controllers/platformoperator_controller.go

```diff
@@ -102,10 +109,51 @@ func (r *PlatformOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	u := updater.NewPlatformOperatorUpdater(r.Client)
+	defer func() {
+		if err := u.Apply(ctx, po); err != nil {
+			log.Error(err, "failed to update status")
+		}
+	}()
+
+	u.UpdateStatus(
+		updater.SetPhase(platformopenshiftiov1alpha1.PhaseFailing),
+		updater.EnsureCondition(metav1.Condition{
+			Type:    platformopenshiftiov1alpha1.TypeInstalled,
+			Status:  metav1.ConditionUnknown,
+			Reason:  platformopenshiftiov1alpha1.ReasonInstalling,
+			Message: "",
+		}),
+	)
+
+	rc, err := handleCatalogConnection(catalogSourceAddress)
+	if err != nil {
+		log.Error(err, "failed to connect to the requisite catalog")
+		u.UpdateStatus(
+			updater.SetPhase(platformopenshiftiov1alpha1.PhaseFailing),
+			updater.EnsureCondition(metav1.Condition{
+				Type:    platformopenshiftiov1alpha1.TypeInstalled,
+				Status:  metav1.ConditionFalse,
+				Reason:  platformopenshiftiov1alpha1.ReasonCatalogUnreachable,
+				Message: err.Error(),
+			}),
+		)
+		return ctrl.Result{}, err
+	}
+
 	log.Info("listing bundles from context")
-	it, err := r.RegistryClient.ListBundles(ctx)
+	it, err := rc.ListBundles(ctx)
 	if err != nil {
 		log.Error(err, "failed to list bundles in the platform operators catalog source")
+		u.UpdateStatus(
+			updater.SetPhase(platformopenshiftiov1alpha1.PhaseFailing),
+			updater.EnsureCondition(metav1.Condition{
+				Type:    platformopenshiftiov1alpha1.TypeInstalled,
+				Status:  metav1.ConditionFalse,
+				Reason:  platformopenshiftiov1alpha1.ReasonCatalogUnstable,
```

Comment:
I think having `Unstable` for when `ListBundles` fails versus `Unreachable` when setting up the client fails is nice attention to detail 👍 

I'm wondering if we can be even more specific though? `ReasonCatalogListBundlesFailed` is even more clear, but may leak too much of the implementation details into the API?

Reply:
Maybe we could do something slightly more vague that `ReasonCatalogListBundlesFailed` in the form of `ReasonCatalogBundleQueryFailed`? That abstracts a bit of the implementation details out of it but conveys the same message.